src/util/sandbox.cpp

```diff
@@ -385,66 +401,76 @@ QString Sandbox::migrateOldSettings() {
+            "\n\n"
+            "If you do not want to grant Mixxx access click Cancel on the file picker."
+        )
+    );
     QString result = QFileDialog::getExistingDirectory(
-            nullptr,
-            title,
-            oldPath);
-    if (result != oldPath) {
-        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate "
-                   "old settings from"
-                << oldPath << "User selected" << result;
+        nullptr,
+        title,
+        legacySettingsPath);
+    if (result != legacySettingsPath) {
+        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate old settings from"
+                << legacySettingsPath << "User selected" << result;
         return sandboxedPath;
     }
 
-    // Sandbox::askForAccess cannot be used here because it depends on settings being
-    // initialized. There is no need to store the bookmark anyway because this is a
-    // one time process.
-#ifdef __APPLE__
+    // Check if Mixxx process is sandboxed
+    QString sandboxContainerId = QProcessEnvironment::systemEnvironment().value("APP_SANDBOX_CONTAINER_ID", "");
+
     CFURLRef url = CFURLCreateWithFileSystemPath(
-            kCFAllocatorDefault, QStringToCFString(oldPath), kCFURLPOSIXPathStyle, true);
+            kCFAllocatorDefault, QStringToCFString(legacySettingsPath), kCFURLPOSIXPathStyle, true);
     if (url) {
         CFErrorRef error = NULL;
-        CFDataRef bookmark = CFURLCreateBookmarkData(
+        if (sandboxContainerId != "") {
+            // Request permissions to old sandboxed settings path in sandbox and move the directory
+            CFDataRef bookmark = CFURLCreateBookmarkData(
                 kCFAllocatorDefault,
                 url,
                 kCFURLBookmarkCreationWithSecurityScope,
                 nil,
                 nil,
                 &error);
-        CFRelease(url);
-        if (bookmark) {
-            QFile oldSettings(oldPath);
-            if (oldSettings.rename(sandboxedPath)) {
-                qInfo() << "Sandbox::migrateOldSettings: Successfully "
-                           "migrated old settings from"
-                        << oldPath << "to new path" << sandboxedPath;
+            CFRelease(url);
+            if (!bookmark) {
+                QFile oldSettings(legacySettingsPath);
+                if (oldSettings.rename(sandboxedPath)) {
+                    qInfo() << "Sandbox::migrateOldSettings: Successfully "
+                            "migrated old settings from"
+                            << legacySettingsPath << "to new path" << sandboxedPath;
+                } else {
+                    qWarning() << "Sandbox::migrateOldSettings: Failed to migrate "
+                                "old settings from" << legacySettingsPath
+                                << "to new path" << sandboxedPath;
+                }
+                CFRelease(bookmark);
             } else {
```

Comment:
Does this ever get triggered with the CFRelease executed unconditionally before this?

Reply:
CFRelease is for the URL and this is the bookmark. Not sure how they behave together.

Reply:
Oh, I misread the code. Nevermind